integration/helper_test.go

```diff
@@ -543,6 +602,146 @@ func setupRestoreFiles(compressionType string, withPlugin bool) {
+	customData := "here is some data\n"
+
+	customData += strings.Repeat("a", dataLength)
+	customData += "here is some data\n"
+
+	if _, err := gzipf.Write([]byte(customData)); err != nil {
+		Fail(fmt.Sprintf("%v", err))
+	}
+	gzipf.Close()
+	f.Close()
+}
+
+func createOidFile(fname string, content string) {
+	fOid, err := os.Create(fname)
+	if err != nil {
+		Fail(fmt.Sprintf("Could not create %s: %v", fname, err))
+	}
+	defer fOid.Close()
+
+	if _, err := fOid.WriteString(content); err != nil {
+		Fail(fmt.Sprintf("Could not write to %s: %v", fname, err))
+	}
+}
+
+func createCustomTOCFile(fname string, dataLength int) {
+	customTOC := fmt.Sprintf(`dataentries:
+  1:
+    startbyte: 0
+    endbyte: 18
+  2:
+    startbyte: 18
+    endbyte: %[1]d
+  3:
+    startbyte: %[1]d
+    endbyte: %d
+`, dataLength+18, dataLength+18+18)
+	fToc, err := os.Create(fname)
+	if err != nil {
+		Fail(fmt.Sprintf("%v", err))
+	}
+
+	if _, err = fToc.WriteString(customTOC); err != nil {
+		Fail(fmt.Sprintf("%v", err))
+	}
+
+	fToc.Close()
+}
+
+/*
+Tests with skip files and the one with flag --on-error-continue
+require a bit more complicated setup, do different setup function.
+Returns file name list which must be deleted when done.
+*/
+func setupRestoreWithSkipFiles(oid int, withPlugin bool) []string {
+	dataLength := 128*1024 + 1
+
+	ret := []string{}
+
+	fileName := dataFileFullPath
+	if withPlugin {
+		fileName = examplePluginTestDataFile
+	}
+	if oid > 0 {
+		fileName = fileName + fmt.Sprintf("_%d", oid)
+	}
+
+	createDataFile(fileName, dataLength)
+	ret = append(ret, fileName)
+
+	// Write oid file
+	createOidFile(restoreOidFile, "1,0\n1,1\n1,2\n")
+	ret = append(ret, restoreOidFile)
+
+	pipename := fmt.Sprintf("%s_%d_0", pipeFile, 1)
+	err := unix.Mkfifo(pipename, 0700)
+	if err != nil {
+		Fail(fmt.Sprintf("%v", err))
+	}
+
+	createCustomTOCFile(tocFile, dataLength)
+	ret = append(ret, tocFile)
+
+	skipFile := fmt.Sprintf("%s_skip_%d", pipeFile, 1)
+	err = exec.Command("touch", skipFile).Run()
+	Expect(err).ToNot(HaveOccurred())
+
+	ret = append(ret, skipFile)
+	return ret
+}
+
+func doTestSkipFiles(oid int, withPlugin bool, args []string) {
+	filesToDelete := setupRestoreWithSkipFiles(oid, withPlugin)
+	for _, f := range filesToDelete {
+		defer func(filename string) {
+			os.Remove(filename)
+		}(f)
+	}
+
+	By("Create restore command")
+	helperCmd := exec.Command(gpbackupHelperPath, args...)
+	err := helperCmd.Start()
+	Expect(err).ToNot(HaveOccurred())
+
+	// Block here until gpbackup_helper finishes (cleaning up pipes)
+	err = helperCmd.Wait()
+	Expect(err).ToNot(HaveOccurred())
+	for _, i := range []int{1, 2, 3} {
+		currentPipe := fmt.Sprintf("%s_%d_0", pipeFile, i)
+		Expect(currentPipe).ToNot(BeAnExistingFile())
+	}
+
+	By("Check in logs that batches were not restored")
+
+	homeDir := os.Getenv("HOME")
+	helperFiles, _ := path.Glob(path.Join(homeDir, "gpAdminLogs/gpbackup_helper_*"))
```

Comment:
why not
```suggestion
	helperFiles, _ := filepath.Glob(filepath.Join(homeDir, "gpAdminLogs/gpbackup_helper_*"))
```
as below in function assertNoErrors?

Reply:
It would be much more productive if you tell why your variant is better. I took it from the same project which appears to have different ways. I have no idea "why not".

Reply:
reduce diff